2.segment_images/scripts/0.segment_nuclei.py

```diff
@@ -24,82 +24,60 @@
 from cellpose import core, models
 from skimage import io
 
-# check if in a jupyter notebook
-try:
-    cfg = get_ipython().config
-    in_notebook = True
-except NameError:
-    in_notebook = False
-
-# Get the current working directory
 cwd = pathlib.Path.cwd()
 
 if (cwd / ".git").is_dir():
     root_dir = cwd
-
 else:
     root_dir = None
     for parent in cwd.parents:
         if (parent / ".git").is_dir():
             root_dir = parent
             break
+sys.path.append(str(root_dir / "utils"))
+from arg_parsing_utils import check_for_missing_args, parse_args
+from notebook_init_utils import bandicoot_check, init_notebook
 
-# Check if a Git root directory was found
-if root_dir is None:
-    raise FileNotFoundError("No Git root directory found.")
+root_dir, in_notebook = init_notebook()
+
+image_base_dir = bandicoot_check(
+    pathlib.Path("/home/lippincm/mnt/bandicoot").resolve(), root_dir
+)
 
 
 # ## parse args and set paths
 
 # If if a notebook run the hardcoded paths.
 # However, if this is run as a script, the paths are set by the parsed arguments.
 
-# In[2]:
+# In[ ]:
 
 
 if not in_notebook:
-    print("Running as script")
-    # set up arg parser
-    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
-
-    parser.add_argument(
-        "--patient",
-        type=str,
-        help="Patient ID to use for the segmentation",
+    args = parse_args()
+    window_size = args["window_size"]
+    clip_limit = args["clip_limit"]
+    well_fov = args["well_fov"]
+    patient = args["patient"]
+    check_for_missing_args(
+        well_fov=well_fov,
+        patient=patient,
+        window_size=window_size,
+        clip_limit=clip_limit,
     )
```

Comment:
It could be nearing the time to use a more sophisticated CLI argument manager for this work. I can't tell for sure, but it might make things easier to manage and reduce code duplication. Generally this block seems to be duplicated many times. Would it be possible to have one notebook/module call the others to reduce the number of times you need to regather the arguments?